daemon/graphdriver/driver.go

```diff
@@ -130,8 +128,8 @@ func getBuiltinDriver(name, home string, options []string, uidMaps, gidMaps []id
 }
 
 // New creates the driver and initializes it at the specified root.
-func New(root string, options []string, uidMaps, gidMaps []idtools.IDMap) (driver Driver, err error) {
-	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
+func New(root string, driverName string, options []string, uidMaps, gidMaps []idtools.IDMap) (driver Driver, err error) {
+	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), driverName} {
```

Comment:
If you are going to pass in the driver name here why not go one step further and just pass in `[]string` and let the caller determine what the name of the environment variable should be.